app/auth/ValidatedActions.scala

```diff
@@ -0,0 +1,39 @@
+package auth
+
+import configuration.GraphQLConfiguration
+import controllers.routes
+import graphql.codegen.GetTransferAgreement.{getTransferAgreement => gta}
+import play.api.mvc.{Action, AnyContent, Request, Result}
+
+import scala.concurrent.ExecutionContext
+
+abstract class ValidatedActions() extends TokenSecurity {
+  implicit val ec: ExecutionContext
+  val graphqlConfiguration: GraphQLConfiguration
+
+  private val getTransferAgreementClient = graphqlConfiguration.getClient[gta.Data, gta.Variables]()
+
+  private def agreed(field: Option[Boolean]) = field.isDefined && field.get
+
+  private def isTransferAgreementComplete(ta: gta.GetTransferAgreement): Boolean = {
```

Comment:
Ah, I think I misunderstood how this was going to work from the API PR. I thought we were just fetching the transfer agreement to see if it existed, rather than expecting the frontend to make a judgement about whether the transfer agreement was complete.

I think this logic should be in the API, because the API's responsible for the integrity of the data and should determine whether or not a transfer agreement is "complete" or not. Other clients might need to work this out, e.g. an admin site or reporting tool.

And there will be similar decisions in future, which will be much easier to handle if they all happen in the consignment API code. For example, the export client will need to know whether the consignment is ready for export, and it shouldn't have to fetch all the consignment data to determine that.

Sorry, I know I'm asking you to change a lot of work that you've already done. I really think it will make things simpler to manage in future!

Reply:
I agree actually and also, the definition of what "complete" means may change if we ever get optional things in the agreement. 
I've moved the  logic to the API and added an `isComplete` parameter to the response and the front end is checking this.

Reply:
That sounds good!

Just one thing on the [query](https://github.com/nationalarchives/tdr-generated-graphql/pull/4) - should we have a separate query for checking whether the transfer agreement is complete? One of the biggest advantages of GraphQL is that you only query what you need, but that query forces the frontend to get lots of info that it _doesn't_ need. It's not such a big deal for that particular query because it's just a list of boolean fields, but it would be good to set a precedent of only querying what you need for when we have more expensive nested queries.

Reply:
That's a very good point. I've put a new query in the generated graphql project https://github.com/nationalarchives/tdr-generated-graphql/pull/6 and updated this PR to use it.